File: server/src/main/kotlin/com/evgenltd/financemanager/pricing/record/pricing-item.kt
package com.evgenltd.financemanager.pricing.record

import java.math.BigDecimal
import java.util.UUID

data class PricingItemRecord(
    val id: UUID?,
    val name: String = "",
    var category: String = "",
    var unit: String = "",
    var defaultQuantity: BigDecimal = BigDecimal.ONE,
)

data class PricingItemFilter(
    val name: String? = null,
    val category: String? = null,
    val unit: String? = null,
    val page: Int = 0,
    val size: Int = 20,
)

File: web/src/types/pricing-item.ts
import { Reference } from '@/types/common/reference'

export type PricingItemReference = Reference

export interface PricingItem {
  id?: string
  name: string
  category: string
  unit: string
  defaultQuantity: number
}

export interface PricingItemFilter {
  name?: string
  category?: string
  unit?: string
  page?: number
  size?: number
}
